### PR TITLE
iII & rabin2 -II: show file info only

### PR DESCRIPTION
**Pull request by kazarmy**

### CODE REVIEW

File: libr/core/cmd_info.c [hunk 1]
```diff
@@ -268,34 +269,47 @@ static int bin_is_executable(RBinObject *obj){
...
 				r_cons_printf (",\"bin\":");
 			}
 			r_core_bin_info (core, R_CORE_BIN_ACC_INFO, mode, va, NULL, NULL);
 		}
 		if (mode == R_CORE_BIN_JSON && array == 0) {
-			r_cons_printf ("}\n");
+			if (!file_only) {
+				r_cons_printf ("}");
+			}
+			r_cons_printf ("\n");
```
Comment (radare) [inline libr/core/cmd_info.c]: use r_cons_newline()

File: libr/core/cmd_info.c [hunk 2]
```diff
@@ -268,34 +269,47 @@ static int bin_is_executable(RBinObject *obj){
...
+		if (!file_only && bin_is_executable (obj)) {
 			if ((mode & R_CORE_BIN_JSON)) {
 				r_cons_printf (",\"bin\":");
 			}
 			r_core_bin_info (core, R_CORE_BIN_ACC_INFO, mode, va, NULL, NULL);
 		}
 		if (mode == R_CORE_BIN_JSON && array == 0) {
-			r_cons_printf ("}\n");
+			if (!file_only) {
+				r_cons_printf ("}");
```
Comment (radare) [inline libr/core/cmd_info.c]: r_cons_strcat to avoid the unnecessary format string thing

Comment (radare) [reply libr/core/cmd_info.c]: feel free to change all the other calls to optimize this

File: libr/core/cbin.c [hunk 3]
```diff
@@ -817,7 +817,7 @@ static int bin_info(RCore *r, int mode) {
 			}
 		}
 		if (IS_MODE_JSON (mode)) {
-			r_cons_printf ("}");
+			r_cons_printf ("}\n");
```
Comment (radare) [inline libr/core/cbin.c]: those newlines where there when printing multiple outputs at once so the json looks in one line. maybe this newline thing should go somewhere else

**Comment by radare** [comment]
it is really necessary to have -II, instead of just showing this info in -I ?didnt tested the PR but looks to me like an unnecessary option.

also, dont merge branches, always rebase

**Comment by kazarmy** [comment]
> it is really necessary to have -II, instead of just showing this info in -I ?

yeah showing the info in -I is better

> also, dont merge branches, always rebase

ok

**Comment by radare** [comment]
can you rebase? and squash?

**Comment by kazarmy** [comment]
> can you rebase?

there's a merge conflict now due to one of my commits ... probably easier (and more proper) to just close this PR and open a new one for https://github.com/radare/radare2/pull/10727#issuecomment-405177593.

**Pull request closed without merging**